### PR TITLE
Ts shuffleboard input

### PR DESCRIPTION
**Pull request by tharunS123**
I solved the sinking between shuffleboard and code issue with Shuffleboard text input feature.

### CODE REVIEW

File: src/main/java/team/gif/robot/Globals.java [hunk 1]
```diff
@@ -3,4 +3,5 @@
 // Should be used sparingly
 public abstract class Globals {
     public static boolean indexerEnabled = true;
+    public static boolean inputValue;
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Globals.java]: Is this necessary? Looks like this is declared twice, once here as a boolean and again in robot.java as a double.

File: src/main/java/team/gif/robot/Robot.java [hunk 2]
```diff
@@ -18,8 +23,19 @@
     public static final boolean isCompBot = false;
 
     private Command m_autonomousCommand;
-
     private RobotContainer m_robotContainer;
+    public static Pigeon m_pigeon = null;
+    public static NetworkTableEntry examplentry;
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]: Misspelling
```suggestion
    public static NetworkTableEntry exampleShuffleboardEntry;
```

File: src/main/java/team/gif/robot/Robot.java [hunk 3]
```diff
@@ -18,8 +23,19 @@
...
+
+    // T.S: Creating an new tab in shuffleboard.
+    ShuffleboardTab tab = Shuffleboard.getTab("FRC2022 test");
+    //public double inputValue;
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]: Delete commented line

File: src/main/java/team/gif/robot/Robot.java [hunk 4]
```diff
@@ -18,8 +23,19 @@
...
+    public ShuffleboardInput inputCommand;
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]: clarify variable name
```suggestion
    public ShuffleboardInput exampleShuffleboardEntryCommand;
```

File: src/main/java/team/gif/robot/Robot.java [hunk 5]
```diff
@@ -18,8 +23,19 @@
...
+
+    // TS: the value of the something what is changing,(Example PID control).
+    private static double inputSink;
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]:
```suggestion
    private static double exampleShuffleboardEntrySyncValue;
```

File: src/main/java/team/gif/robot/Robot.java [hunk 6]
```diff
@@ -18,8 +23,19 @@
...
+    // TS: the value is getting the getEntry number
+    public static double inputValue = inputSink;
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]:
```suggestion
    public static double exampleShuffleboardValue = exampleShuffleboardEntrySyncValue;
```

File: src/main/java/team/gif/robot/Robot.java [hunk 7]
```diff
@@ -30,6 +46,24 @@ public void robotInit() {
         // Instantiate our RobotContainer.  This will perform all our button bindings, and put our
         // autonomous chooser on the dashboard.
         m_robotContainer = new RobotContainer();
+        // TS: getting the submit button when you click the commend.
+        inputCommand = new ShuffleboardInput();
+
+        m_pigeon = new Pigeon();
+        m_pigeon.addToShuffleboard("Shuffleboard", "Pigeon");
+
+        // TS: Commented the getEntry method and implemented a button.
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]: I don't understand this comment

File: src/main/java/team/gif/robot/Robot.java [hunk 8]
```diff
@@ -30,6 +46,24 @@ public void robotInit() {
...
+        examplentry = tab.add("Example Input",inputValue)
+                .getEntry();
+        // TS: displaying the submib button.
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]:
```suggestion
        // Add the Example Input Submit button to the Shuffleboard.
```

File: src/main/java/team/gif/robot/Robot.java [hunk 9]
```diff
@@ -30,6 +46,24 @@ public void robotInit() {
...
+        tab.add("Command", inputCommand);
+        // TS: setting the getEntry value to (Example PID control).
+        examplentry.setDouble(inputSink);
+
+        /**
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/Robot.java]: delete this commented section

File: src/main/java/team/gif/robot/commands/ShuffleboardInput.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
+package team.gif.robot.commands;
+
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import team.gif.robot.Robot;
+
+public class ShuffleboardInput extends CommandBase {
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/commands/ShuffleboardInput.java]: Change this command to:
exampleShuffleboardEntryCommand

File: src/main/java/team/gif/robot/commands/ShuffleboardInput.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+    public ShuffleboardInput(){}
+
+
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+        Robot.inputValue = Robot.examplentry.getDouble(Robot.inputValue);
```
Comment (billknopfjr) [inline src/main/java/team/gif/robot/commands/ShuffleboardInput.java]: comment these 2 lines to explain why we have to perform both a get and a set

**Review by billknopfjr — CHANGES_REQUESTED** [review]
Take a look a the changes I suggested, make modifications, and then lets test again.

File: src/main/java/team/gif/robot/Globals.java (hunk 1 above)
Comment (tharunS123) [reply src/main/java/team/gif/robot/Globals.java]: No, we don't need it. I forget to delete that line.

**Review by billknopfjr — COMMENTED** [review]
Changes look good. Let's do a final test on Monday and if it all looks good, I will merge it into main.

**Review by billknopfjr — APPROVED** [review]

**Pull request merged**